pcsx2/SIO/Memcard/MemoryCardFile.cpp

```diff
@@ -636,6 +637,50 @@ void FileMcd_Reopen(std::string new_serial)
 	FileMcd_EmuOpen();
 }
 
+void FileMcd_Swap()
+{
+	if (MemcardBusy::IsBusy())
+	{
+		Host::AddIconOSDMessage("MemoryCardSwap", ICON_PF_MEMORY_CARD, TRANSLATE_SV("MemoryCardSwap", "Memory cards are busy, cannot swap right now."));
+		return;
+	}
+
+	const std::string card1Filename = Host::GetStringSettingValue("MemoryCards", "Slot1_Filename");
+	const std::string card2Filename = Host::GetStringSettingValue("MemoryCards", "Slot2_Filename");
+
+	// Copy each McdOptions to local memory
+	Pcsx2Config::McdOptions firstSlot = EmuConfig.Mcd[0];
+	Pcsx2Config::McdOptions secondSlot = EmuConfig.Mcd[1];
+
+
+	if (!firstSlot.Enabled || !secondSlot.Enabled ||
+		card1Filename.empty() || card2Filename.empty())
+	{
+		Host::AddIconOSDMessage("MemoryCardSwap", ICON_PF_MEMORY_CARD, TRANSLATE_SV("MemoryCard", "Both slots must have a card selected to swap.."));
+		return;
+	}
+
+	// Swap them
+	Host::AddIconOSDMessage("MemoryCardSwap", ICON_PF_MEMORY_CARD, TRANSLATE_SV("MemoryCard", "Swapping Memory Cards.."));
```

Comment:
Ditto on trailing periods. Removing the extra period or making this an ellipsis, either would be fitting.